modules/axhal/src/platform/riscv64_qemu_virt/console.rs

```diff
@@ -1,14 +1,28 @@
+use memory_addr::VirtAddr;
+
+use crate::mem::virt_to_phys;
+
 /// Writes a byte to the console.
 pub fn putchar(c: u8) {
-    #[allow(deprecated)]
-    sbi_rt::legacy::console_putchar(c as usize);
+    sbi_rt::console_write_byte(c);
+}
+
+/// Writes bytes to the console from input u8 slice.
+pub fn write_bytes(bytes: &[u8]) {
+    sbi_rt::console_write(sbi_rt::Physical::new(
+        bytes.len(),
+        virt_to_phys(VirtAddr::from_ptr_of(bytes.as_ptr())).as_usize(),
+        0,
+    ));
 }
 
-/// Reads a byte from the console, or returns [`None`] if no input is available.
-pub fn getchar() -> Option<u8> {
-    #[allow(deprecated)]
-    match sbi_rt::legacy::console_getchar() as isize {
-        -1 => None,
-        c => Some(c as u8),
-    }
+/// Reads bytes from the console into the given mutable slice.
+/// Returns the number of bytes read.
+pub fn read_bytes(bytes: &mut [u8]) -> usize {
+    sbi_rt::console_read(sbi_rt::Physical::new(
```

Comment:
Limit buffer size, like 256, to e SBI seems to disable IRQs, we should not read too many bytes.

Reply:
should we use a warning here or simply use a assertion

Reply:
Just use `bytes.len().min(MAX_READ_SIZE)`

Reply:
> Just use `bytes.len().min(MAX_READ_SIZE)`

ok, get it. But should we give user some feedback like a warning like bytes length exceeds the `MAX_READ_SIZE`

Reply:
since we return number of bytes to read in the return value.
i think no warning or other feedback should be given